test/performance/latency/LatencyTestTypes.cpp

```diff
@@ -24,55 +24,58 @@
 using namespace eprosima::fastrtps;
 using namespace eprosima::fastrtps::rtps;
 
+const std::string LatencyDataType::type_name_ = "LatencyType";
+
+bool LatencyDataType::compare_data(
+        const LatencyType& lt1,
+        const LatencyType& lt2) const
+{
+    if (lt1.seqnum != lt2.seqnum)
+    {
+        return false;
+    }
+
+    return 0 == memcmp(lt1.data, lt2.data, buffer_size_);
+}
+
 bool LatencyDataType::serialize(void*data,SerializedPayload_t* payload)
 {
     LatencyType* lt = (LatencyType*)data;
 
     memcpy(payload->data, &lt->seqnum, sizeof(lt->seqnum));
-    const auto size = static_cast<uint32_t>(lt->data.size());
-    memcpy(payload->data + 4, &size, sizeof(size));
-
-    //std::copy(lt->data.begin(),lt->data.end(),payload->data+8);
-    memcpy(payload->data + 8, lt->data.data(), lt->data.size());
-    payload->length = (uint32_t)(8+lt->data.size());
+    memcpy(payload->data + 4, lt->data, buffer_size_);
+    payload->length = 4 + buffer_size_;
     return true;
 }
 
 bool LatencyDataType::deserialize(SerializedPayload_t* payload,void * data)
 {
+    // Payload members endianness matches local machine
     LatencyType* lt = (LatencyType*)data;
-    memcpy(&lt->seqnum, payload->data, sizeof(lt->seqnum));
-    uint32_t size;
-    memcpy(&size, payload->data+4, sizeof(size));
-    std::copy(payload->data+8,payload->data+8+size,lt->data.begin());
+    lt->seqnum = *reinterpret_cast<uint32_t*>(payload->data);
+    std::copy(payload->data+4,payload->data+4+buffer_size_,lt->data);
     return true;
 }
 
-std::function<uint32_t()> LatencyDataType::getSerializedSizeProvider(void* data)
+std::function<uint32_t()> LatencyDataType::getSerializedSizeProvider(void*)
 {
-    return [data]() -> uint32_t
+    uint32_t size = m_typeSize;
+    return [size]() -> uint32_t
     {
-        LatencyType *tdata = static_cast<LatencyType*>(data);
-        uint32_t size = 0;
-
-        size = (uint32_t)(sizeof(uint32_t) + sizeof(uint32_t) + tdata->data.size());
-
         return size;
     };
 }
 
 void* LatencyDataType::createData()
 {
-
-    return (void*)new LatencyType();
+    return (void*)new uint8_t[(4 + buffer_size_ + 3) / 4 * 4];
```

Comment:
```suggestion
    return (void*)new uint8_t[m_typeSize];
```

Reply:
Done in 759afcbf9cb2c567b30b3b3475617648754d5895